tools/sklearn/main_macros.xml

```diff
@@ -104,6 +101,77 @@ def get_X_y(params, file1, file2):
   return X, y
   </token>
 
+  <token name="@GET_SEARCH_PARAMS_FUNCTION@">
+def get_search_params(params_builder, in_pipeline=False):
+
+    search_params = {}
+
+    for ep in params_builder['estimator_param_set']:
+        if ep['estimator_param'].strip() == '':
+            continue
+        lst = ep['estimator_param'].split(":")
+        if len(lst) >2:
+            exec lst[2].strip()
+        literal = lst[1].strip().replace("__sq__","'")\
```

Comment:
You can avoid this replacing  by allowing the parameter to use these characters with a sanatizer.

Reply:
Good point. Thanks.